### PR TITLE
[incubator-kie-issues-1105] Polymorphic json typing causes long delays in first request

### PR DESCRIPTION
**Pull request by elguardian**
Closes https://github.com/apache/incubator-kie-issues/issues/1105

### CODE REVIEW

File: kie-server-parent/kie-server-api/pom.xml [hunk 1]
```diff
@@ -38,11 +38,17 @@
   </properties>
 
   <dependencies>
+    <dependency>
+        <groupId>org.projectlombok</groupId>
+        <artifactId>lombok</artifactId>
+        <version>1.18.32</version>
+        <scope>provided</scope>
+    </dependency>
```
Comment (martinweiler) [inline kie-server-parent/kie-server-api/pom.xml:46]: I don't think we should / need to add lombok here, we could just remove all the related annotations from the dummy classes

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java [hunk 2]
```diff
@@ -347,6 +354,10 @@ public TypeDeserializer buildTypeDeserializer(DeserializationConfig config, Java
             typer2 = typer2.init(JsonTypeInfo.Id.CLASS, null);
             typer2 = typer2.inclusion(JsonTypeInfo.As.WRAPPER_OBJECT);
             deserializeObjectMapper.setDefaultTyping(typer2);
+            // this avoids the use of default typing
+            if(!defaultTypingEnabled) {
+                deserializeObjectMapper.deactivateDefaultTyping();
```
Comment (martinweiler) [inline kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java:359]: if we disable default typing here, wouldn't it make sense to wrap lines 327-356 in a `if(defaultTypingEnabled)` clause?

File: kie-server-parent/kie-server-api/src/test/java/com/example/manyrules/JSONMarshallerLargeTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assume.assumeThat;
+
+public class JSONMarshallerLargeTest {
+
+    private Logger logger = LoggerFactory.getLogger(JSONMarshallerLargeTest.class);
+
+    @Test(timeout=5000L)
+    public void testLargeNumberOfClasses() throws IOException {
+        assumeThat(JSONMarshaller.isDefaultTypingActivate(), is(false));
```
Comment (martinweiler) [inline kie-server-parent/kie-server-api/src/test/java/com/example/manyrules/JSONMarshallerLargeTest.java:27]: We should run the test by default with `org.kie.server.json.defaultTyping.enabled=false` and not skip it

File: kie-server-parent/kie-server-api/pom.xml (hunk 1 above)
Comment (elguardian) [reply kie-server-parent/kie-server-api/pom.xml:46]: removed

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java (hunk 2 above)
Comment (elguardian) [reply kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java:359]: done

File: kie-server-parent/kie-server-api/src/test/java/com/example/manyrules/JSONMarshallerLargeTest.java (hunk 3 above)
Comment (elguardian) [reply kie-server-parent/kie-server-api/src/test/java/com/example/manyrules/JSONMarshallerLargeTest.java:27]: done

**Review by martinweiler — APPROVED** [review]
Great work @elguardian

**Pull request merged**